Prepare the auto-sizes module as a standalone plugin

joemcgill opened the pull request:
## Summary

This updates the auto-sizes module feature to prepare for merging into trunk. Included in this change:

- Updates the hook function names
- Adds a plugin readme.txt file to the module folder
- Fixes the Plugin header name

See: #904 

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

joemcgill committed:
Update hook function names

joemcgill committed:
Add plugin readme.txt

joemcgill committed:
Fix Plugin header name.

felixarntz commented on modules/images/auto-sizes/readme.txt:
```diff
@@ -0,0 +1,53 @@
+
+== Description ==
+
+This plugin implements the HTML spec for auto sizes for lazy-loaded images.
```
Should we maybe include a mention of `sizes="auto"` here or elsewhere? While "Auto sizes" is a decent name, there's a good chance a developer will have a better idea of what this does when seeing `sizes="auto"`, as only that clarifies it's about the "sizes" _attribute_. "Auto sizes" could also have to do with e.g. the generated image sizes or other things.

I don't want to get hung up on that, but I think at least one mention of the attribute would be helpful in clarifying. For instance:
```suggestion
This plugin implements the HTML spec for `sizes="auto"` for lazy-loaded images.
```

joemcgill replied:
I'm happy to go with this. Thanks!

felixarntz reviewed (approved):
@joemcgill LGTM, just one readme docs recommendation.

joemcgill committed:
Update plugin description

Specifically reference `sizes="auto"` in the description, rather than "auto sizes".

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

mukeshpanchal27 commented on modules/images/auto-sizes/load.php:
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Module Name: Auto Sizes for Lazy-loaded Images.
+ * Module Name: Auto Sizes for Lazy-loaded Images
```
It's not a blocker, but in the [CODEOWNERS file](https://github.com/WordPress/performance/blob/feature/auto-sizes/.github/CODEOWNERS#L57), we have `Auto-sizes for Lazy-Loaded Images` specified, and the module name differs from this designation. Should we consider correcting this discrepancy for consistency?

joemcgill replied:
Good observation! Looking at the official spec and various references from early browser implementers, I see this both hyphenated and not, but it's good for us to be consistent, so I've updated all references to be hyphenated in [6870baf](https://github.com/WordPress/performance/pull/922/commits/6870baf52050e5e1240db347cd7d55670b4d58bb).

mukeshpanchal27 reviewed (approved):
Thank you, @joemcgill, for the PR! I've provided one non-blocking feedback. Aside from that, everything appears to be in good from my perspective.

joemcgill committed:
Consistently use hyphenated 'auto-sizes'
